File: server/src/SavedMessage.js
const mongoose = require('mongoose');

const savedMessageSchema = new mongoose.Schema({
  userId: { type: String, required: true, index: true },
  content: {
    ciphertext: { type: String, required: true },
    nonce: { type: String, required: true },
  },
  label: { type: String }, // Optional title for the saved message
  createdAt: { type: Date, default: Date.now }
});

savedMessageSchema.index({ userId: 1, createdAt: -1 });

module.exports = mongoose.model('SavedMessage', savedMessageSchema);

File: flutter_app/lib/models/saved_message.dart
class SavedMessage {
  final String id;
  final Map<String, dynamic> encryptedContent;
  final String? label;
  final DateTime createdAt;

  SavedMessage({
    required this.id,
    required this.encryptedContent,
    this.label,
    required this.createdAt,
  });

  factory SavedMessage.fromJson(Map<String, dynamic> json) {
    return SavedMessage(
      id: json['_id'],
      encryptedContent: Map<String, dynamic>.from(json['content']),
      label: json['label'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}